aten/src/ATen/core/op_registration/op_registration_test.cpp

```diff
@@ -39,6 +39,29 @@ struct MockKernel final : OperatorKernel {
   bool* called_;
 };
 
+TEST(OperatorRegistrationTest, whenRegisterOperatorWithArgToBeIntType_thenPass) {
+  const auto& functionSchema = torch::jit::parseSchema("_test::dummy(Tensor dummy, float arg0, int arg1) -> ()");
+  const auto& dispatchKeyExtractor = c10::DispatchKeyExtractor::make(functionSchema);
+  auto scalarTypeIndex = dispatchKeyExtractor.getScalarTypeIndex();
+  ASSERT_TRUE(scalarTypeIndex.has_value());
+  EXPECT_EQ(scalarTypeIndex.value(), 2);
+}
+
+TEST(OperatorRegistrationTest, whenRegisterOperatorWithMultipleArgToBeIntType_thenPassAndReturnFirstIndex) {
```

Comment:
Once we change this to ScalarType, it is not "the first ScalarType" we want but the one that is in the correct constellation with DeviceType and Layout. And if we have multiple of those triplets, we should probably throw an error rather than taking the first one because we don't support that.